webpack.unmix.js

```diff
@@ -0,0 +1,384 @@
+        if (name.lastIndexOf('?') > 0) {
+          // querystring version
+          name = name.substring(0, name.lastIndexOf('?'));
+        } else {
+          // hash in filename version
+          const extname = path.extname(name);
+          let basename = name.substring(0, name.lastIndexOf(extname));
+          basename = basename.substring(0, basename.lastIndexOf('.'));
+
+          name = `${basename}${extname}`;
+        }
+
+        manifest[name] = path.join(compiler.options.output.publicPath, asset.name);
+      });
+
+      fs.writeFileSync(this.fileName, JSON.stringify(manifest, null, 2));
+    });
+  }
+}
+
+//#endregion
+
+//#region entrypoints and output
+const entry = {
+  'app': [
+    './resources/assets/app.ts',
+    './resources/assets/less/app.less',
+  ],
+};
+
+const coffeeReactComponents = [
+  'artist-page',
+  'beatmap-discussions',
+  'beatmap-discussions-history',
+  'beatmapset-page',
+  'changelog-build',
+  'changelog-index',
+  'comments-index',
+  'comments-show',
+  'mp-history',
+  'modding-profile',
+  'profile-page',
+  'admin/contest',
+  'contest-entry',
+  'contest-voting',
+];
+
+const tsReactComponents = [
+  'account-edit',
+  'beatmaps',
+  'chat',
+  'friends-index',
+  'groups-show',
+  'news-index',
+  'news-show',
+  'notifications-index',
+  'scores-show',
+  'store-bootstrap',
+];
+
+for (const name of coffeeReactComponents) {
+  entry[`react/${name}`] = [resolvePath(`resources/assets/coffee/react/${name}.coffee`)];
+}
+
+for (const name of tsReactComponents) {
+  entry[`react/${name}`] = [resolvePath(`resources/assets/lib/${name}.ts`)];
+}
+
+const output = {
+  filename: outputFilename('js/[name]', 'js'),
+  path: resolvePath('public/assets'),
+  publicPath: '/assets/',
+};
+
+//#endregion
+
+//#region plugin list
+const plugins = [
+  new webpack.ProvidePlugin({
+    $: 'jquery',
+    _: 'lodash',
+    Cookies: 'js-cookie',
+    d3: 'd3', // TODO: d3 is fat and probably should have it's own chunk
+    jQuery: 'jquery',
+    moment: 'moment',
+    React: 'react',
+    ReactDOM: 'react-dom',
+    Turbolinks: 'turbolinks',
+  }),
+  new webpack.DefinePlugin({
+    'process.env.DOCS_URL': JSON.stringify(process.env.DOCS_URL || 'https://docs.ppy.sh'),
+    'process.env.PAYMENT_SANDBOX': JSON.stringify(paymentSandbox),
+    'process.env.SHOPIFY_DOMAIN': JSON.stringify(process.env.SHOPIFY_DOMAIN),
+    'process.env.SHOPIFY_STOREFRONT_TOKEN': JSON.stringify(process.env.SHOPIFY_STOREFRONT_TOKEN),
+  }),
+  new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/), // don't add moment locales to bundle.
+  new MiniCssExtractPlugin({
+    filename: outputFilename('css/[name]', 'css'),
+  }),
+  new CopyPlugin({
+    patterns: [
+      { from: 'resources/assets/build/locales', to: outputFilename('js/locales/[name]') },
+      { from: 'node_modules/moment/locale', to: outputFilename('js/moment-locales/[name]') },
+    ],
+  }),
+  new Manifest({ fileName: 'public/assets/manifest.json'}),
```

Comment:
should this be `${output.path}/manifest.json` :thinking: (or at least use `resolvePath`) (also karma doesn't build anything and thus it fails creating manifest because missing directory?)